src/Client/Connection.cpp

```diff
@@ -985,6 +985,10 @@ void Connection::initBlockInput()
         }
 
         block_in = std::make_unique<NativeReader>(*maybe_compressed_in, server_revision);
+        /// proton: starts
```

Comment:
What we can do is copying the Connection.h / Connection.cpp and Cluster.h/.cpp etc to `ClickHouse` folder, moving forward, we vendor ClickHouse instead of creating Proton folder and vendor our own protocol code.

Reply:
Everything in Clickhouse folder, we can update / modify at different paces as we go to make it work with ClickHouse external tables.

Reply:
I think the downside is that it makes things harder when we need to port changes from the ClickHouse repo.

Reply:
This is a good concern, vendor which worth more thinking. If we vendor ourself to a different folder when we change our own protocol and leaving the main structure keeping aligned with the CH community makes the porting things easier. Let's keep as it is for now.